.github/workflows/ci.yaml

```diff
@@ -86,3 +86,13 @@ jobs:
         name: failsafe-reports-${{ matrix.profile.name }}
         path: ./**/target/failsafe-reports/
         if-no-files-found: ignore
+  typo-check:
+    name: Typo Check
+    # only run on pull requests because of security reasons
+    # we shouldn't trust external actions for builds within the repository
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Check typos
+        uses: crate-ci/typos@v1.22.4
```

Comment:
I think we could add some links(possibly pinned to concrete tag) for:
* https://github.com/crate-ci/typos/blob/master/docs/reference.md for ".typos.toml".
* https://github.com/crate-ci/typos/blob/master/docs/github-action.md (here ?)